fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/advancement/FabricCriterionRegistry.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.advancement;
+
+import net.minecraft.advancement.criterion.Criterion;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.mixin.object.builder.CriteriaAccessor;
+
+/**
+ * Allows registering advancement criterions for triggers.
+ *
+ * <p>A registered criteria can be retrieved through
+ * {@link net.minecraft.advancement.criterion.Criteria#getById(Identifier)}.</p>
+ *
+ * @see net.minecraft.advancement.criterion.Criteria
+ */
+public final class FabricCriterionRegistry {
```

Comment:
I'd say you have a registry of `criteria` but you register `criterion`.

So I'd say `FabricCriteriaRegistry` is probably a more accurate name

Reply:
No, we say "block registry" than "blocks registry". We use singular names.

Reply:
Honestly I'd probably just call it FabricCriteria though we do tend to use the word Registry elsewhere so I guess it's fine

Reply:
Just `CriterionRegistry` would work here. Everything doesn't need a Fabric prefix.

Reply:
I think the current name is fine, the FabricPrefix isnt needed here, but its also not wrong. In most cases the Fabric prefix is only needed when there is an existing mc class with the same name.